server/container_create.go

```diff
@@ -648,7 +648,12 @@ func (s *Server) setupSeccomp(specgen *generate.Generator, profile string) error
 		return nil
 	}
 	if profile == seccompRuntimeDefault || profile == seccompDockerDefault {
-		return seccomp.LoadProfileFromStruct(&s.seccompProfile, specgen)
+		linuxSpecs, err := seccomp.LoadProfileFromConfig(s.seccompProfile, specgen.Config)
+		if err != nil {
+			return err
+		}
+		specgen.Config.Linux.Seccomp = linuxSpecs
+		return nil
 	}
 	if !strings.HasPrefix(profile, seccompLocalhostPrefix) {
```

Comment:
The TrimPrefix below should also use seccompLocalhostPrefix.

Reply:
Yes, changed as well.